charts/istio-endpoints/templates/sidecar.yaml

```diff
@@ -6,6 +6,12 @@ metadata:
   name: {{ include "istio-endpoints.fullname" . }}-proxy-config
   labels:
     {{- include "istio-endpoints.labels" . | nindent 4 }}
+  {{- with .Values.sidecar.annotations }}
+  annotations:
```

Comment:
I don't see anything specifically wrong with this, so it's fine. I am not sure that Argo Sync Waves are exactly the right answer to the problem though.

Reply:
just noting this for future reference: since kubernetes operates under the expectation of eventual consistency, trying to enforce the ordering of specific resources may cause other issues/side effects. going to try this and see if it's okay for now, or if we need a different approach.